src/lib/cbfs.c

```diff
@@ -58,7 +58,10 @@ enum cb_err _cbfs_boot_lookup(const char *name, bool force_ro,
 			   RO CBFS would have been caught when building the mcache in cbfs_get
 			   boot_device(). (Note that TOCTOU_SAFETY implies !NO_CBFS_MCACHE.) */
 			assert(cbd == vboot_get_cbfs_boot_device());
-			die("TODO: set metadata_hash to RW metadata hash here.\n");
+			if (!CONFIG(VBOOT)
+			    || vb2api_get_metadata_hash(vboot_get_context(), &metadata_hash)
```

Comment:
Please explain in the commit message why this is TOCTOU-safe.

Reply:
it is an upstream change